### DISCUSSION TITLE
Expand the use of the ShrinkEveryNTurns parameter

### DISCUSSION THREAD

**Discussion by mawilson** [BattlesnakeOfficial/feedback#208] [Feature Requests]
I was just running some games to test some Sinkhole-specific code I'd written to model the hazard spawns. In [this Custom-source game](https://play.battlesnake.com/g/7bc8b4c3-ae34-4ae0-9998-a745d600a1c0/) I was surprised to see the sinkhole boundaries growing every 10 turns, instead of 20 like in the Sinkhole battleground. Currently, I don't see a way for my snake to dynamically determine how quickly the sinkhole will grow.

Would it make sense to expand the use of the royale 'shrinkEveryNTurns' attribute to be provided in Sinkhole games to tell us how often the sinkhole will expand? This parameter could also be provided in other contexts, like in Hazard Spiral for how often the spiral grows, Hazard Scatter for how often it scatters, Healing Pools for how often the pools disappear, etc. Maybe this is a poor fit for this attribute & another should be introduced, or maybe these type of settings could be included in the map name, e.g., 'sinkholes10' or 'sinkholes20'.

**Comment by chris-bsnake** [comment]
So effectively that is what we are doing right now.  For example here is the game data provided to the snake for the custom Sinkholes map. The map is using the `ruleset.settings.royal.shrinkEveryNTurns` value to determine when to grow the hazard area.  

```
  "game": {
    "id": "04d75fcf-42a2-4fd2-8191-55db6f956864",
    "ruleset": {
      "name": "standard",
      "version": "v1.1.15",
      "settings": {
        "foodSpawnChance": 15,
        "minimumFood": 1,
        "hazardDamagePerTurn": 10,
        "hazardMap": "",
        "hazardMapAuthor": "",
        "royale": {
          "shrinkEveryNTurns": 20
        },
        "squad": {
          "allowBodyCollisions": false,
          "sharedElimination": false,
          "sharedHealth": false,
          "sharedLength": false
        }
      }
    },
    "map": "sinkholes",
    "timeout": 500,
    "source": "custom"
  },
```
One thing we want to do this fall is refactor the `settings` to be more generic.  The `royale` subgroup is an artifact of Royale originally being a game mode, but now that it is a map, i think we rework how that data is represented, though that may involve us doing a major update to the Battlesnake API.

**Reply by mawilson** [reply]
Oh, I feel silly, I do see that in the game data now. Could've sworn I had looked & it wasn't in there. Thanks for the clarification! I agree that making it more generic probably makes more sense in the future.
